numpy/core/src/npysort/timsort.cpp

```diff
@@ -982,24 +977,22 @@ static inline int
 resize_buffer_(string_buffer_<Tag> *buffer, npy_intp new_size)
 {
     using type = typename Tag::type;
+
+    type *new_pw;
+
     if (new_size <= buffer->size) {
         return 0;
     }
 
-    if (NPY_UNLIKELY(buffer->pw == NULL)) {
-        buffer->pw = (type *)malloc(sizeof(type) * new_size * buffer->len);
-    }
-    else {
-        buffer->pw = (type *)realloc(buffer->pw,
-                                     sizeof(type) * new_size * buffer->len);
-    }
-
+    new_pw =
+            (type *)realloc(buffer->pw, sizeof(type) * new_size * buffer->len);
```

Comment:
```suggestion
    new_pw = (type *)realloc(buffer->pw, sizeof(type) * new_size * buffer->len);
```
Not worth it for one character.